### ISSUE TITLE
[Sprint] Upgrade ARIMA to ARIMAX and Improve Auto-Tuning

### ISSUE THREAD

**Issue by ankit02327** [ankit02327/stock-price#112]
As part of the Algorithmic & ML Sprint, we need to upgrade our ARIMA implementation (`backend/algorithms/optimised/arima/arima.py`) to utilize exogenous variables (technical indicators) and improve the robustness of its hyperparameter tuning.

Currently, the `fit(X, y)` method completely discards the `X` feature matrix (which contains valuable technical indicators). Upgrading the `statsmodels` implementation to use `exog=X` will change this from a simple ARIMA to a much more powerful ARIMAX model. Furthermore, the manual grid search in `_find_optimal_order` silently suppresses all errors (`except: continue`), which masks convergence failures.

**Tasks:**
- [ ] Upgrade the `ARIMA` fitting logic to accept exogenous variables (`exog=X`), utilizing the technical indicators for prediction.
- [ ] Refactor the `predict` method to accept and pass the future exogenous variables `exog=X`.
- [ ] Fix the silent exception catching in `_find_optimal_order` to explicitly log `ConvergenceWarnings` instead of ignoring them.
- [ ] Consider replacing the manual grid search loop with the `pmdarima` library, which provides a highly robust and optimized `auto_arima` function.

**Comment by ankit02327** [comment]
1. **Implement ARIMAX:** Update the initialization of `ARIMA(series, order=self.order)` to `ARIMA(endog=series, exog=X, order=self.order)` inside the `fit()` method. 
2. **Update Forecasting:** Ensure the `predict` and `predict_with_confidence` methods pass the new `X` data (representing future technical indicators) as the `exog` parameter to `self.fitted_model.forecast(steps=steps, exog=X)`.
3. **Fix Error Suppression:** In `_find_optimal_order`, replace `except:` with `except Exception as e:` and use the `logging` module to log failures as debug or warning messages, so developers can see if the model is failing to converge on specific (p,d,q) orders.

**Comment by ankit02327** [comment]
```py
# 1. Update fit method to use exogenous variables (ARIMAX)
def fit(self, X: np.ndarray, y: np.ndarray) -> 'ModelInterface':
    self.validate_input(X, y)
    series = pd.Series(y)
    
    # ... optimal order logic ...
    
    # Pass X as exogenous variables
    self.model = ARIMA(endog=series, exog=X, order=self.order, seasonal_order=self.seasonal_order)
    self.fitted_model = self.model.fit()
    # ... rest of method

# 2. Update predict methods
def predict(self, X: np.ndarray, steps: int = 1) -> np.ndarray:
    self.validate_input(X)
    
    # Ensure X length matches steps
    if len(X) != steps:
        raise ValueError(f"For ARIMAX, length of exogenous variables X ({len(X)}) must match steps ({steps})")
        
    forecast = self.fitted_model.forecast(steps=steps, exog=X)
    # ... return forecast

# 3. Fix silent error suppression
try:
    model = ARIMA(endog=series, exog=X, order=(p, d, q))
    fitted = model.fit()
    if fitted.aic < best_aic:
        best_aic = fitted.aic
        best_order = (p, d, q)
except Exception as e:
    logger.debug(f"ARIMA order {(p,d,q)} failed to converge: {str(e)}")
    continue
```
